node/src/commands/destroy.js

```diff
@@ -1,20 +1,36 @@
 const DeployUtils = require('../lib/deploy-utils');
+const { options } = require('../config/constants');
+const _ = require('lodash');
+const logger = require('../lib/logger');
+
+const { PROJECT_ID, ENVIRONMENT_NAME, REQUIRES_APPROVAL, ARCHIVE_AFTER_DESTROY } = options;
+
+const assertEnvironmentExists = environment => {
+  if (!environment) {
+    throw new Error(`Could not find an environment with the name ${options[ENVIRONMENT_NAME]}`);
+  }
+};
 
 const destroy = async options => {
   const deployUtils = new DeployUtils();
 
-  const environment = await deployUtils.getEnvironment(options.environmentName, options.projectId);
+  logger.info('Waiting for deployment to start...');
+  const environment = await deployUtils.getEnvironment(options[ENVIRONMENT_NAME], options[PROJECT_ID]);
   let status;
 
-  if (environment) {
-    const deployment = await deployUtils.destroyEnvironment(environment);
-    status = await deployUtils.pollDeploymentStatus(deployment.id);
+  assertEnvironmentExists(environment);
 
-    if (options.archiveAfterDestroy) await deployUtils.archiveIfInactive(environment.id);
-  } else {
-    throw new Error(`Could not find an environment with the name ${options.environmentName}`);
+  const requiresApproval = eval(options[REQUIRES_APPROVAL]);
```

Comment:
Please don't use `eval` ever ever ever never ever.
If you want to convert a string to boolean you can replace this one with `JSON.parse` or even just `str === 'true'`

Reply:
![](https://media0.giphy.com/media/xEpTspH9hGwHS/giphy.gif?cid=5a38a5a26q4xt1br8ofj1vhqtguclv6hu3ftyevqsz8ncs6u&rid=giphy.gif)

Reply:
🤣 

I know it's known as unsafe, but felt that since I validate its parameter already then it's fine 🤷 
Anyways:
0cab956